Kernel/Syscall.cpp

```diff
@@ -135,7 +139,13 @@ void syscall_handler(TrapFrame* trap)
 
     // Apply a random offset in the range 0-255 to the stack pointer,
     // to make kernel stacks a bit less deterministic.
-    auto* ptr = (char*)__builtin_alloca(get_fast_random<u8>());
+    // Since this is very hot code, request random data in chunks instead of
+    // one byte at a time. This is a noticeable speedup.
+    if (g_random_byte_buffer_offset == RandomByteBufferSize) {
```

Comment:
Hmm, doesn't this mean we'll not fill the buffer with randomness until the 256th syscall?

Reply:
The main slowdown (I think) is `AESCipher` allocating a new `AESCipherBlock` when it's constructed, pulling data out is then allocation-free, so I'd say this is the right approach here

Reply:
Whoops!